src/common/xoa-updater.js

```diff
@@ -0,0 +1,427 @@
+import assign from 'lodash/assign'
+import Client from 'jsonrpc-websocket-client'
+import eventToPromise from 'event-to-promise'
+import forEach from 'lodash.foreach'
+import makeError from 'make-error'
+import map from 'lodash/map'
+import { EventEmitter } from 'events'
+import {
+  xoaConfiguration,
+  xoaRegisterState,
+  xoaTrialState,
+  xoaUpdaterLog,
+  xoaUpdaterState
+} from 'store/actions'
+
+// ===================================================================
+
+const states = [
+  'disconnected',
+  'updating',
+  'upgrading',
+  'upToDate',
+  'upgradeNeeded',
+  'registerNeeded',
+  'error'
+]
+
+// ===================================================================
+
+export function isTrialRunning (trial) {
+  return (trial && trial.end && Date.now() < trial.end)
+}
+
+export function exposeTrial (trial) {
+  // We won't suggest trial if any trial is running now, or if premium was enjoyed in any past trial
+  return !(trial && (isTrialRunning(trial) || trial.plan === 'premium'))
+}
+
+export function blockXoaAccess (xoaState) {
+  let block = xoaState.stata === 'untrustedTrial'
+  // STARTER, ENTERPRISE, PREMIUM ONLY
+  block = block || xoaState.state === 'ERROR'
+  // END of STARTER, ENTERPRISE, PREMIUM ONLY
+  return block
+}
+
+function getCurrentUrl () {
+  if (typeof window === 'undefined') {
+    throw new Error('cannot get current URL')
+  }
+  return String(window.location)
+}
+
+function adaptUrl (url, port = null) {
+  const matches = /^http(s?):\/\/([^\/:]*(?::[^\/]*)?)(?:[^:]*)?$/.exec(url)
+  if (!matches || !matches[2]) {
+    throw new Error('current URL not recognized')
+  }
+  return 'ws' + matches[1] + '://' + matches[2] + '/api/updater'
+}
```

Comment:
Errr..."not" be necessary ?...Ok for JsonRpcWebSocketClient features, but is this harming overkill ? There could be another client in the future for any reason, or not, but...